app/grandchallenge/algorithms/templates/algorithms/job_detail.html

```diff
@@ -12,7 +12,7 @@
 {% load civ %}
 
 {% block title %}
-    Algorithm Result - {{ block.super }}
+    {{ object.pk }} - {{ object.algorithm_image.algorithm.title }} - {{ block.super }}
```

Comment:
I think it would be clearer if you prepend either Job or Algorithm Result to the pk. Just doing `{{ object }}` might be the simplest, that will return "Job 1234"